example/src/domain/machinery/Form.js

```diff
@@ -0,0 +1,109 @@
+import { useFieldSnapshot, useFieldValue, useFormField, useNumberFormField, useArrayFormField } from '@kaliber/forms'
+import { Code } from './Code'
+
+export function FormValues({ form }) {
+  const formState = useFieldSnapshot(form)
+  return <Code value={formState} indent />
+}
+
+export function FormTextInput({ field, label }) {
+  const { name, state, eventHandlers } = useFormField(field)
+  return <InputBase type='text' {...{ name, label, state, eventHandlers }} />
+}
+
+export function FormNumberInput({ field, label }) {
+  const { name, state, eventHandlers } = useNumberFormField(field)
+  return <InputBase type='text' {...{ name, label, state, eventHandlers }} />
```

Comment:
type zou hier `number` moeten zijn.

Reply:
Weet ik, maar is expres text om te zorgen dat je de `number` validatie kunt testen, zal er een comment bij plaatsen. Of misschien is het een beter idee om er even een ander component naast te zetten die dit duidelijker maakt.